### PR TITLE
rapidcheck: refactor `postInstall` to install all extras

### PR DESCRIPTION
**Pull request by tsmanner**
RapidCheck's extras were not accessible through the package, with the exception of boost_test, which was being copied directly into the core include directory.  [RapidTest's documentation](https://github.com/emil-e/rapidcheck/blob/master/doc/boost_test.md#usage) for that module says to add the `extras/boost_test/include` directory to your include path, but the nixpkgs derivation for it puts the file in `$out/include/rapidtest`.

> This support is available through the `extras/boost_test` module. You can either directly add the `extras/boost_test/include` directory to your include path or link against the `rapidcheck_boost_test` target in your `CMakeLists.txt`. You can then simply `#include <rapidcheck/boost_test.h>`. Note that `rapidcheck/boost_test.h` needs to be included after any Boost Test headers.

This commit replaces the copy of that one header with an `rsync` of all files, except `CMakeLists.txt`s, from `$src/extras` into `$out/extras` so that the output path matches what the rapidtest documentation expects.  Accessing all extras is now possible by adding `"${pkgs.rapidtest}/extras/<extra>/include"` to your list of include paths.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tsmanner** [comment]
I haven't done any cross platform testing of this, only confirmed that it works in WSL Debian with gcc 9.3.0.

One thing that isn't clear to me yet is whether or not the `boost` and `gmock` `cpp` files are compiled as part of the normal build, or if they'll need to get their own derivations to get them optionally included in the `rapidtest.a` that is built by default.

File: pkgs/development/libraries/rapidcheck/default.nix [hunk 1]
```diff
@@ -11,10 +11,12 @@ stdenv.mkDerivation rec {
...
   };
 
-  nativeBuildInputs = [ cmake ];
+  nativeBuildInputs = [ cmake rsync ];
 
+  # Install the extras headers
   postInstall = ''
-    cp ../extras/boost_test/include/rapidcheck/boost_test.h $out/include/rapidcheck
+    cd $src
+    rsync --filter="- **CMakeLists.txt" -acRv extras $out
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/rapidcheck/default.nix:19]: Can you please use mv or cp?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/rapidcheck/default.nix (hunk 1 above)
Comment (tsmanner) [reply pkgs/development/libraries/rapidcheck/default.nix:19]: I originally had tried using `cp`, but you can't filter the files being copied.  Considering that it already depends on `cmake` I was thinking that a dependency on `rsync` at build time would be preferable to having installed the extras includes with the `CMakeLists.txt` files still lying around in there.  The thing I had tried was a recursive `cp` followed by an `rm **/CMakeLists.txt`, but the destination files were read only, so I switched.  If that approach would work by changing the file permissions in `$out` before doing the `rm`, then I'm happy to do that.

I played around with `builtins.filterSource` as well, but it isn't allowed to refer to store paths, so I was left wondering if there was a way to use a nix function to filter that directory, but wasn't sure what to try.

Comment (tsmanner) [reply pkgs/development/libraries/rapidcheck/default.nix]: I tried with the `chmod +w` and `rm` for the `CMakeLists.txt` files and it is producing the same result as the `rsync`, just in a more roundabout way.

Comment (SuperSandro2000) [reply pkgs/development/libraries/rapidcheck/default.nix]: It is preferable to only use rsync if there is no other way. 4 Lines bash instead is fine for me.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 106376` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rapidcheck</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 106376` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rapidcheck</li>
  </ul>
</details>

**Pull request merged**